### ISSUE TITLE
Simple query running too many Redis Get according to New Relic

### ISSUE THREAD

**Issue by othiagocruz** [balderdashy/sails-redis#89]
Hey guys thanks for the past inputs, continuing here the discussion started on https://github.com/balderdashy/waterline/issues/1276

@Ryanc1256 I was told you could have some information on this question, could you take a look?

I forgot to add the header of the report, it follows:

![screen shot 2016-01-19 at 6 28 26 pm](https://cloud.githubusercontent.com/assets/1930027/12431071/866ec9f2-beda-11e5-975d-5f9c16285cdf.png)

Also here's the model being queryed:
```
module.exports = {

  attributes: {
    text        : { type: 'string' },
    position    : { type: 'string' },
    category    : { model: 'Category' },
    owner       : { model: 'User' },
    published   : { type: 'boolean' },
    metadata    : { type: 'json' }
  },
  afterUpdate: function(updatedRecord, cb) {
    Content.destroy({position: updatedRecord.position, category: updatedRecord.category, sort: 'createdAt DESC', published: false, skip: 10}).exec(function(err){
      if (err) return cb(err);
      cb();
    });
  },
  afterCreate: function(newlyInsertedRecord, cb) {
    Content.destroy({position: newlyInsertedRecord.position, category: newlyInsertedRecord.category, sort: 'createdAt DESC', published: false, skip: 10}).exec(function(err){
      Content.find({position: newlyInsertedRecord.position, category: newlyInsertedRecord.category}).exec(function (err, relatedContent) {
        if (err) return cb(err);
        if(relatedContent) {
          relatedContent.map(function(item){
            item.published = (newlyInsertedRecord.id == item.id);
            item.save();
          });
          cb();
        }
      });
    });

  }
};
```
Supposedly, afterUpdate and afterCreate are not related to the query monitored on new relic since the report its from the API of the app, the GUI its on another report.

**Comment by Ryanc1256** [comment]
@othiagocruz yeah, so was going to look at it when you posted it on the other repo, just though to wait until you did it here...


Okay so the reason why there are so many "get" request's is that when you do a ".find" it will find all the keys relating to your model, so "Content" and do single ".get" requests to grab the data out of them, then it will search the json string side the keys to check if your ".find" query matches the model it has just gotten and sterilized

So the way to get around it is, defining a primary key atm it's the default which is "id", so i would search by the "id" instead of anything else, that would reduce the amount of ".get" requests...

So for example...

If i had 200 Content models for example
If i went "Content.find({position: 1})"
It would call ".get" 200 times and sterilize the keys to see if position is equal to "1" or whatever I inserted into the find query. 

What I would do is...

Content.find( 32 )

this will just find the models with the primary key, but since you don't really want that functionality in your case, you can't really bring down the "get" requests down.


It kind of goes it issue #80 too, I'm trying to fix it but it's gonna take sometime.

**Comment by othiagocruz** [comment]
@Ryanc1256 I see, so its a recursion issue really. Thanks that been helpful, I think I could try to query the category model to find out its content and let the orm do the relations to see if I get less gets, will experiment on that and on the approach you mentioned!

**Comment by sailsbot** [comment]
Thanks for posting, @othiagocruz.  I'm a repo bot-- nice to meet you!

The issue queue in this repo is for verified bugs with documented features.  Unfortunately, we can't leave some other types of issues marked as "open".  This includes bug reports about undocumented features or unofficial plugins, as well as feature requests, questions, and commentary about the core framework. Please review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) for details.

If you're here looking for help and can't find it, visit [StackOverflow](http://stackoverflow.com), our [Google Group](https://groups.google.com/forum/#!forum/sailsjs) or our [chat room](https://gitter.im/balderdashy/sails?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge).  But please do not let this disrupt future conversation in this issue! I am only marking it as "closed" to keep organized.

Thanks so much for your help!

> If I am mistaken, and this issue is about a _critical bug with a documented feature_, please accept my sincerest apologies-- I am but a humble bot working to make GitHub a better place.  If you open a new issue, a member of the core team will take a look ASAP.

**Comment by sailsbot** [comment]
Thanks for posting, @othiagocruz.  I'm a repo bot-- nice to meet you!

The issue queue in this repo is for verified bugs with documented features.  Unfortunately, we can't leave some other types of issues marked as "open".  This includes bug reports about undocumented features or unofficial plugins, as well as feature requests, questions, and commentary about the core framework. Please review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) for details.

If you're here looking for help and can't find it, visit [StackOverflow](http://stackoverflow.com), our [Google Group](https://groups.google.com/forum/#!forum/sailsjs) or our [chat room](https://gitter.im/balderdashy/sails?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge).  But please do not let this disrupt future conversation in this issue! I am only marking it as "closed" to keep organized.

Thanks so much for your help!

> If I am mistaken, and this issue is about a _critical bug with a documented feature_, please accept my sincerest apologies-- I am but a humble bot working to make GitHub a better place.  If you open a new issue, a member of the core team will take a look ASAP.

**Comment by irlnathan** [comment]
@Ryanc1256  I see you've discovered sailsbot doesn't like the "question" label :simple_smile:  Really appreciate you responding to @othiagocruz.   There's some new stuff in the contribution guide that's worth a gander. see -- https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md#ii-issue-contributions

**Comment by Ryanc1256** [comment]
@irlnathan yeah, stupid bot lol! the thing is, it is valid question and is backed by a valid issue that i had mentioned above and that's why I'm trying to keep it around, as it's a valid... But then again it's loosely relating to #80 so "eh", we could close it, since it's relating to other issue, but I'm just keeping it because there are people out there that are bound to be trying to find out the same thing...

**Comment by irlnathan** [comment]
@Ryanc1256 Understood...this is something we're trying to solve with the new contribution guide. Github does a great job with back links, and linking to an open issue or pr that matches the guidelines from the closed one is the best way to make sure we see questions we can help with. However, if a question is a special exception, you can always not label it a question, and sailsbot will leave it alone :)

**Comment by sailsbot** [comment]
Thanks for posting, @othiagocruz.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
